src/pod.rs

```diff
@@ -31,9 +31,9 @@ pub fn from_bytes<T: Pod>(data: &[u8]) -> Result<(&T, &[u8])> {
     let size = mem::size_of::<T>();
     let tail = data.get(size..).ok_or(())?;
     let ptr = data.as_ptr();
-    if (ptr as usize) % mem::align_of::<T>() != 0 {
-        return Err(());
-    }
+    // if (ptr as usize) % 8 != 0 {
```

Comment:
I was unable to read certain structures from my local dyld shared cache because of this alignment requirement. My understanding is that the alignment should be that of the primitive types within the structure, but it appears even that isnt correct as only an alignment of 2 seemed to work.

Reply:
Which structures and fields in particular? If they are specific to dyld shared cache, then those fields should use types such as `U32Bytes` instead of `U32`. If it's Mach-O structures for example, then it may be that the dyld shared cache isn't aligned them the same way as normal files, and the `unaligned` feature of this crate is required.

Reply:
I'm getting it when reading the slide info:
```rust
                let slide_info_file_offset = self.slide_info_file_offset.get(endian);
                let version = data
                    .read_at::<U32<E>>(slide_info_file_offset)
                    .read_error("Invalid slide info file offset size or alignment")?
                    .get(endian);
                match version {
                    5 => {
                        let slide = data
                            .read_at::<macho::DyldCacheSlideInfo5<E>>(slide_info_file_offset)
                            .read_error("Invalid dyld cache slide info size or alignment")?;
                        let page_starts_offset = slide_info_file_offset
                            .checked_add(mem::size_of::<macho::DyldCacheSlideInfo5<E>>() as u64)
                            .unwrap();
                        let page_starts = data
                            .read_slice_at::<U16<E>>(
                                page_starts_offset,
                                slide.page_starts_count.get(endian) as usize,
                            )
                            .read_error("Invalid page starts size or alignment")?;
                        Ok(Some(DyldCacheSlideInfoSlice::V5(slide, page_starts)))
                    }
                    _ => todo!("handle other dyld_cache_slide_info versions"),
                }
```
which has this format:
```rust
pub struct DyldCacheSlideInfo5<E: Endian> {
    pub version: U32<E>,   // currently 5
    pub page_size: U32<E>, // currently 4096 (may also be 16384)
    pub page_starts_count: U32<E>,
    reserved1: u32,
    pub value_add: U64<E>,
}
```

Reply:
If the `slide_info_file_offset` you're seeing only has 32-bit alignment then `value_add` needs to change to `U64Bytes`. And if it only has 8-bit alignment then you need to change the `U32` as well.

Reply:
Looks like it was actually just the alignment of the `reserved1` field that was the issue.

Reply:
Sounds like you're already using the `unaligned` feature then (which only affects `U32` and `U64`, not `u32`), so the other fields technically should be changed too so that it works without that feature.

Reply:
I don't think I am, unless it's the default? I have just added the dependency with `cargo add object` and changed the path to the local dir I have object checked out to.

Reply:
Yes it's the default.